### PR TITLE
[WIP] Switch to the mariadb connector from the mysql one.

### PR DESCRIPTION
**Pull request by kiwiroy**
Switch from `com.mysql.cj.jdbc.Driver` to `org.mariadb.jdbc.Driver`.

## References

#4745 

## Checklist

Happy to work on these as required.

- [x] Change in CHANGELOG.md described
- [x] Tests created for changes (modified original)
- [x] Manually tested changed features in running JabRef ([travis](https://travis-ci.org/kiwiroy/jabref/builds/504417225))
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [x] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -98,7 +98,7 @@ dependencies {
     compile 'org.apache.pdfbox:pdfbox:2.0.14'
     compile 'org.apache.pdfbox:fontbox:2.0.14'
     compile 'org.apache.pdfbox:xmpbox:2.0.14'
-
+    compile group: 'org.mariadb.jdbc', name: 'mariadb-java-client', version: '2.4.0'
```
Comment (tobiasdiez) [inline build.gradle]: Please group this with the other db-related dependencies (mysql/postgre).
Can the mysql dependency be deleted now?

Comment (kiwiroy) [reply build.gradle]: Will do.

**Comment by LinusDietz** [comment]
Hey @kiwiroy, thank you for this contribution!
After some minutes of comparing the JDBC drivers, I'm not sure why mariadb is the obvious way to go. So please add more reasoning, both here in the comments, as well as a condensed version in an architecture decision record here https://github.com/JabRef/jabref/tree/master/docs/adr

This will help us to judge upon the benefits and consequences of this change.

**Comment by Siedlerchr** [comment]
I'm curious: Would it be possible to have both connectors e.g. MYSQL and MARIADB together so that a user can choose between them or does this create conflics?

**Comment by kiwiroy** [comment]
Just some comments here first:

I'm unsure if this is relevant, but this [blog] has *some* background information.
MariaDB provide a comprehensive [feature comparison], though that is mainly
server features rather than client. As far as possible they maintain protocol
[compatibilty] for clients.

For the linked issue (#4745) the MySQL connector does not connect to the PAM
enabled server despite [claims]. The MySQL connector documentation mention a
configuration property [authenticationPlugins] which is a *"Comma-delimited list
of classes that implement com.mysql.cj.protocol.AuthenticationPlugin"*. A PAM
class, I'd expect `dialog`, cannot be seen in the connector either.

com.mysql.cj.protocol.a.authentication

The [MariaDB] implementations include a specific class for this.

This [StackOverflow][stackoverflow-1] thread pointed me to the MariaDB connector
as a replacement that would achieve a connection and the comment in [this one][stackoverflow-2]
suggests that performance will not be an issue.

> The performance in mysql and with the mysql-connectorj was worse in all cases than mariadb

It is possible to have the two resident in the class path [concurrently].

[blog]: https://hackr.io/blog/mariadb-vs-mysql
[feature comparison]: https://mariadb.com/kb/en/library/mariadb-vs-mysql-features/
[compatibility]: https://mariadb.com/kb/en/library/mariadb-vs-mysql-compatibility/
[authenticationPlugins]: https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-configuration-properties.html
[results]: https://dev.mysql.com/doc/search/?d=218&p=1&q=AuthenticationPlugin
[MariaDB]: https://github.com/MariaDB/mariadb-connector-j/tree/master/src/main/java/org/mariadb/jdbc/internal/com/send/authentication
[stackoverflow-1]: https://stackoverflow.com/a/26930596/7456461
[stackoverflow-2]: https://stackoverflow.com/questions/50473450/mariadb-jdbc-driver-not-batching-updates-efficiently-vs-sql-server
[PAM walkthough]: https://mariadb.com/resources/blog/writing-a-mariadb-pam-authentication-plugin/
[claims]: https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-using-pam.html
[serverPAM]: https://dev.mysql.com/doc/refman/8.0/en/pam-pluggable-authentication.html
[concurrently]: https://github.com/MariaDB/mariadb-connector-j/blob/master/documentation/use-mariadb-connector-j-driver.creole#having-mariadb-and-mysql-driver-in-the-same-classpath

**Comment by kiwiroy** [comment]
First [pass](https://github.com/JabRef/jabref/compare/master...kiwiroy:concurrently) at mysql **and** mariadb. Travis test suite [passes](https://travis-ci.org/kiwiroy/jabref/builds/504587880) and local testing of MariaDB connector against the database with PAM auth succeeds, while only changing the dropdown to MySQL results in the `Unable to load authentication plugin 'dialog'.` message.

**Comment by Siedlerchr** [comment]
Yeah cool that it works! That would be best compromise here I think.

**Comment by kiwiroy** [comment]
Right... Should this PR be closed and a new one created - after a little tidy up?

Also, any pointers on how [this passed](https://travis-ci.org/kiwiroy/jabref/builds/504575907) while including [this line](https://github.com/kiwiroy/jabref/compare/51ff33cb61e2^...648c92231c6a#diff-a465c056fb9283b0efd14091f4eb8d66R42).

**Comment by Siedlerchr** [comment]
Hm, that is really odd regarding the test. Try debugging the test locally or add a System.out.println before comparing the values to see the output

You can either reuse this  PR if you want or use the other PR. Whatever you want

**Comment by Siedlerchr** [comment]
@kiwiroy  What is the current status here? It would be really nice to have this inlcuded

**Comment by kiwiroy** [comment]
I’ve had a busy few weeks, will attend to it soon

**Comment by tobiasdiez** [comment]
@kiwiroy any update?

**Comment by LinusDietz** [comment]
Closing due to inactivity. Feel free to reopen it if you have time to work on it.

**Comment by koppor** [comment]
I resolved the conflicts and are going to merge it. I will add an ADR.

We keep only MariaDB. Reason: MySQL Connector/J 8.0 is licensed under GPL. Source: <https://downloads.mysql.com/docs/licenses/connector-j-8.0-gpl-en.pdf>

> Without limiting the foregoing grant of rights under the GPLv2 and additional permission as to separately licensed software, this Connector is also subject to the Universal FOSS Exception, version 1.0, a copy of which is reproduced below and can also be found along with its FAQ at http://oss.oracle.com/licenses/universal-foss-exception

I understand that there is that FOSS exception, but I am not aware of the consequences - and that exception is not commonly known.

**Pull request merged**